runtime/src/parachains.rs

```diff
@@ -274,32 +188,39 @@ decl_module! {
 	pub struct Module<T: Trait> for enum Call where origin: <T as system::Trait>::Origin {
 		/// Provide candidate receipts for parachains, in ascending order by id.
 		#[weight = SimpleDispatchInfo::FixedNormal(1_000_000)]
-		fn set_heads(origin, heads: Vec<AttestedCandidate>) -> Result {
+		pub fn set_heads(origin, heads: Vec<AttestedCandidate>) -> Result {
```

Comment:
the documentation here says that the heads must be in ascending order by ID. the `on_finalize` function of registrar seems to assume that parathreads are at the end of the `DidUpdate` vec (doesn't appear to be checked by this function).

https://github.com/paritytech/polkadot/pull/341/files#diff-6ceafc4b4dd2600a9e5caebc167c0cecR374

Reply:
yeah docs needed updating.

Reply:
perhaps you missed the line 196?

```
active_parachains.sort_by_key(|&(ref id, _)| *id);
```

Reply:
This should be resolved with my latest commit.